src/core/Merger.cpp

```diff
@@ -624,5 +621,15 @@ Merger::ChangeList Merger::mergeMetadata(const MergeContext& context)
             changes << tr("Adding missing icon %1").arg(QString::fromLatin1(customIconId.toRfc4122().toHex()));
         }
     }
+
+    const auto customDataKeys = sourceMetadata->customData()->keys();
+    for (const auto& key : customDataKeys) {
+        if (!targetMetadata->customData()->contains(key)) {
```

Comment:
If the key does exist, we should check to see if the values are the same. If they are not... then I think the best course of action at this time is to overwrite the key with the new value. There is also the case where a key is deleted on the source but will return again once the merge operation completes (because it is still on the target). The problem is custom data does not have a modified time or an operations history.

Reply:
The first implementation of this just overwrote everything, and check for existing key was totally missing. We are overwriting it anyway, right? Of course checking reduces the actual operations.

Reply:
I need to think about this

Reply:
We should store a custom data key/value that records the last time the custom data itself was modified. This would allow us to make better judgement calls on what to do. So we should store a key _LAST_MODIFIED = UTC Date/Time. Lacking this key we assume the source DB is "newer".

If Source is Newer:
* New Key -> Add to Target
* Missing Key -> Delete from Target
* Existing Key -> Replace Value on Target

If Source is Older:
* Don't modify target data

Reply:
This is what KeePass does:

```
foreach(KeyValuePair<string, string> kvp in pdSource.m_dCustomData)
{
    if(bSourceNewer || !m_dCustomData.Exists(kvp.Key))
        m_dCustomData.Set(kvp.Key, kvp.Value);
}
```

Reply:
I agree. This is a very nice approach. I was also thinking storing the last modified time so comparisons can be made.

Reply:
One possible solution is to clone a key with _LAST_MODIFIED value added to the key name. This would double the key/value pairs but then every pair would have its own date/time saved.

Reply:
I think that is overkill. The time state of the whole custom data is sufficient to make decisions.

Reply:
Got it. I'll make the changes.